onnxruntime: Build suddenly fails without concrete error-message

### Steps To Reproduce
Steps to reproduce the behavior:
1. build *onnxruntime* (on nixos-unstable - commit-id "652e92b8064949a11bc193b90b74cb727f2a1405" e.g. for obs-backgroundremoval)

### Build log
https://gist.github.com/SebTM/f4453981849c6caba711675b8e4f2353
https://gist.github.com/SebTM/4f501a1ce608e3a452a6406485326865 (tried update to v1.13.1 - [here](https://github.com/SebTM/nixpkgs/tree/update/onnxruntime_1131))

Maybe (I'm not sure because it still continues building first) this is related to the error in the end:
```
/build/source/onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc: In lambda function:
/build/source/onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc:348:41: error: 'struct dnnl::memory::desc' has no member named 'dims'
  348 |           auto output_shape = output_md.dims();
      |                                         ^~~~
```

### Additional context
Seems to be also not up2date (v1.13.1 latest) and having a history of being unstable/unmaintained (in thepast) and removed from from nixpkgs for that (and readded improved later)

### Notify maintainers
@jonringer 
@puffnfresh 
@ck3d

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.11, NixOS, 23.05 (Stoat), 23.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(user): `""`
 - channels(root): `""`
 - nixpkgs: `/run/nixpkgs`
```

--- a/pkgs/development/libraries/oneDNN/default.nix
+++ b/pkgs/development/libraries/oneDNN/default.nix
@@ -5,13 +5,13 @@
 # https://github.com/oneapi-src/oneDNN#oneapi-deep-neural-network-library-onednn
 stdenv.mkDerivation rec {
   pname = "oneDNN";
-  version = "3.0";
+  version = "2.7.1";
 
   src = fetchFromGitHub {
     owner = "oneapi-src";
     repo = "oneDNN";
     rev = "v${version}";
-    sha256 = "sha256-rhuu2C1etU3IHaLSvUiJKMwgnPpcm1g40xjwDprc2LU=";
+    sha256 = "sha256-HBCuSZkApd/6UkAxz/KDFb/gyX2SI1S2GwgXAXSTU/c=";
   };
 
   outputs = [ "out" "dev" "doc" ];

